model_interfaces/ltm_agent_3.py

```diff
@@ -275,28 +208,6 @@ def reply(self, user_content: str) -> str:
         self.add_to_memory(assistant_message)
         return response
 
-    def completion(self, context: List[dict[str, str]], temperature: float, label: str) -> str:
-        def cost_callback(cost_usd: float):
-            self.costs_usd += cost_usd
-
-        response = ask_llm(
-            context, self.model, temperature=temperature, context_length=None, cost_callback=cost_callback
-        )
-        if _log_prompts:
-            with self.log_lock:
-                self.log_count += 1
-                log_dir = f"./logs/{self.session_id}"
-                os.makedirs(log_dir, exist_ok=True)
-                prompt_file = f"{label}-prompt-{self.log_count}.json"
-                prompt_json = json.dumps(context, indent=2)
-                prompt_path = os.path.join(log_dir, prompt_file)
-                with codecs.open(prompt_path, "w", "utf-8") as fd:
-                    fd.write(prompt_json)
-                completion_path = os.path.join(log_dir, f"{label}-completion-{self.log_count}.txt")
-                with codecs.open(completion_path, "w", "utf-8") as fd:
-                    fd.write(response)
-        return response
-
     def reset_history(self):
```

Comment:
`reset_history` and `reset_all` could be moved mainly to the base class, and then subclasses do
```python
super().reset_history()
# whatever they do in addition
```

Reply:
Yea, there's sill some duplication there that could be removed.